source/D2Common/src/Items/ItemMods.cpp

```diff
@@ -655,59 +655,72 @@ BOOL __stdcall ITEMMODS_UpdateItemWithSkillCharges(D2UnitStrc* pItem, int nSkill
 }
 
 //D2Common.0x6FD928D0 (#10843)
-//TODO: Find names for variables
-int __stdcall D2COMMON_10843_GetByTimeAdjustment(int nAmount, int nTimeOfDay, int nBaseTime, int* a4, int* a5, int* a6)
+int __stdcall ITEMMODS_GetByTimeAdjustment(int nAmount, int nPeriodOfDay, int nBaseTime, int* nItemModPeriodOfDay, int* nItemModMin, int* nItemModMax)
 {
-	int v6;
-	int v7;
-	int v8;
-	int v9;
-	int v10;
+	// Time works in a cycle of 360 units.
+	constexpr int DAY_UNITS = 360;
 
-	v6 = ((nAmount >> 12) & 1023) - 256;
-	v7 = nAmount & 3;
-	v8 = ((nAmount >> 2) & 1023) - 256;
-
-	if (a4)
+	int nTempItemModPeriodOfDay = nAmount & 3;
+	if (nItemModPeriodOfDay != nullptr)
 	{
-		*a4 = v7;
+		*nItemModPeriodOfDay = nTempItemModPeriodOfDay;
 	}
 
-	if (a5)
+	int nTempItemModMin = ((nAmount >> 2) & 1023) - 256;
+	if (nItemModMin != nullptr)
 	{
-		*a5 = v8;
+		*nItemModMin = nTempItemModMin;
 	}
 
-	if (a6)
+	int nTempItemModMax = ((nAmount >> 12) & 1023) - 256;
+	if (nItemModMax != nullptr)
 	{
-		*a6 = v6;
+		*nItemModMax = nTempItemModMax;
 	}
 
-	v9 = nBaseTime - 90 * v7;
-	if (v9 < 0)
+	// Calculate the time difference. Each day period occupies an equal
+	// share of unit in a day. There are 4 periods of day.
+	constexpr int NUM_PERIODS_OF_DAY = 4;
```

Comment:
Could you please rebase and use the newly added `NUM_ENVIRONMENT_PERIODS_OF_DAY` ? (see https://github.com/ThePhrozenKeep/D2MOO/blob/master/source/D2Common/include/D2Environment.h#L13 )